pkg/migration/controllers/migration.go

```diff
@@ -2735,3 +2738,79 @@ func isObjectHashEqual(dynamicClient dynamic.ResourceInterface, u *unstructured.
 	}
 	return false
 }
+
+// getRelatedCRDListWRTGroupAndComponents takes input of AppRegs
+// and find out all CRDs those need to be collected.
+// The CRD list should contain
+//
+//	a. CRD matching with the required groups.
+//	b. CRDs which are related with the CRDs from above list (#a)
+func getRelatedCRDListWRTGroupAndComponents(client *apiextensionsclient.Clientset, ruleset *inflect.Ruleset, crdList *stork_api.ApplicationRegistrationList, resGroups map[string]string) []*stork_api.ApplicationRegistration {
+	filteredCRDList := make([]*stork_api.ApplicationRegistration, 0)
+	reqCategoriesMap := make(map[string]bool)
+	//categoriesToCRDMap := make(map[string][]*stork_api.ApplicationRegistrationList)
+	// find all related categories from the CRDs having same group
+	for _, crd := range crdList.Items {
+		for _, v := range crd.Resources {
+			if _, ok := resGroups[v.Group]; !ok {
+				continue
+			}
+			crdName := ruleset.Pluralize(strings.ToLower(v.Kind)) + "." + v.Group
+			crdvbeta1, err := client.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), crdName, metav1.GetOptions{})
+			if err == nil {
+				for _, category := range crdvbeta1.Spec.Names.Categories {
+					if !slice.ContainsString(catergoriesExcludeList, category, strings.ToLower) {
+						reqCategoriesMap[category] = true
+					}
+				}
+				continue
+			}
+			crdv1, err := client.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), crdName, metav1.GetOptions{})
+			if err == nil {
+				for _, category := range crdv1.Spec.Names.Categories {
+					if !slice.ContainsString(catergoriesExcludeList, category, strings.ToLower) {
+						reqCategoriesMap[category] = true
+					}
+				}
+				continue
+			}
+		}
+	}
+	logrus.Infof("crd categories to include are: %+v", reqCategoriesMap)
+
+	// find all crds matching the group and having related categories
+	for _, tempCRD := range crdList.Items {
+		crd := tempCRD
+		for _, v := range crd.Resources {
+			if _, ok := resGroups[v.Group]; ok {
+				filteredCRDList = append(filteredCRDList, &crd)
+				continue
+			}
+			crdName := ruleset.Pluralize(strings.ToLower(v.Kind)) + "." + v.Group
```

Comment:
nit..nit..I think we can move the statement in the below if condition.

Reply:
Done